lib/puppet/parser/functions/new.rb

```diff
@@ -864,7 +866,7 @@
 | b | The data is in base64 encoding, padding as required by base64 strict is added by default
 | u | The data is in URL safe base64 encoding
 | B | The data is in base64 strict encoding
-| s | The data is a puppet string used verbatim as a binary byte sequence
+| s | The data is a puppet string seen as a binary sequence of bytes with ignored encoding
```

Comment:
There must be a way to get "the encoding that is used by the String", which in Ruby means "whatever" - it is not in our power to change that right here. We also do not have any other platforms to consider atm. When we do, that implementation must be done such that reading a binary byte sequence into a String (a totally stupid thing to do that should be forbidden as Binary should be used, but if you went ahead and did that anyway) it must either use the same style as Ruby and embed the encoding in the string, or be read so that Binary(str, '%s') produces the same byte sequence as was originally read (details how it does that is of no importance but could be reading ISO-8859-1, and then converting with the same encoding).